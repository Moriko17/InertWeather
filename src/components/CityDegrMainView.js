import React from "react";
import "../styles/styles.css";

export default function CityDegrMainView(props) {
  const initData = {
    cityName: "Vostorg",
    temp: 17.3
  };
  let data;
  props.fetchedData ? (data = props.fetchedData) : (data = initData);
  return (
    <div className="city-degr">
      <p className="city">{data.cityName}</p>
      <p className="degr">{data.temp}</p>
    </div>
  );
}
