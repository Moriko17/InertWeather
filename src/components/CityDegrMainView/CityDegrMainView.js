import React from "react";
import "../../styles/styles.css";

export default function CityDegrMainView(props) {
  const initData = {
    cityName: "Vostorg",
    temp: 17.3,
    icon: "01n"
  };
  let data;
  props.fetchedData ? (data = props.fetchedData) : (data = initData);
  const ICON_URL = `https://openweathermap.org/img/wn/${data.icon}@2x.png`;
  return (
    <div className="city-degr">
      <p className="city">{data.cityName}</p>
      <img className="weather-pic" src={ICON_URL} alt="weather pic" />
      <p className="degr">{data.temp} Â°C</p>
    </div>
  );
}
