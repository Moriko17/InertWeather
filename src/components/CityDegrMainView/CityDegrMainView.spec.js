import React from "react";
import renderer from "react-test-renderer";
import CityDegrMainView from "./CityDegrMainView";

describe("CityDegrMainView renders right", () => {
  const fetchedData = {
    cityName: "Paris",
    temp: 17.3,
    icon: "01n"
  };

  it("Renders correctly with no data", () => {
    const tree = renderer.create(<CityDegrMainView />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders correctly with passed data", () => {
    const tree = renderer
      .create(<CityDegrMainView fetchedData={fetchedData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
