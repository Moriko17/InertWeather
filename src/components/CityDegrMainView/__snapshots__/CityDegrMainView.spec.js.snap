// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityDegrMainView renders right Renders correctly with no data 1`] = `
<div
  className="city-degr"
>
  <p
    className="city"
  >
    Vostorg
  </p>
  <img
    alt="weather pic"
    className="weather-pic"
    src="https://openweathermap.org/img/wn/01n@2x.png"
  />
  <p
    className="degr"
  >
    17.3
     °C
  </p>
</div>
`;

exports[`CityDegrMainView renders right Renders correctly with passed data 1`] = `
<div
  className="city-degr"
>
  <p
    className="city"
  >
    Paris
  </p>
  <img
    alt="weather pic"
    className="weather-pic"
    src="https://openweathermap.org/img/wn/01n@2x.png"
  />
  <p
    className="degr"
  >
    17.3
     °C
  </p>
</div>
`;
